common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -778,8 +778,20 @@ public void execute(Runnable task) {
         addTask(task);
         if (!inEventLoop) {
             startThread();
-            if (isShutdown() && removeTask(task)) {
-                reject();
+            if (isShutdown()) {
```

Comment:
Why is this code even caring about `isShutdown()`? It looks like this block should just be deleted. `isShutdown()` is checked by `addTask()`/`offerTask()`. If we added the entry to the queue, then it seems we should commit ourselves to it; attempting to remove it (which is not guaranteed to be successful, since the task may have already run) seems weird.

If this `isShutdown()` check was atomic with `startThread()`, I could sort of understand the removal since we know we didn't start the thread, but as it is it just seems like useless effort.